nixos/modules/system/activation/activation-script.nix

```diff
@@ -94,6 +94,18 @@ in
 
     };
 
+    environment.usrbinenv = mkOption {
+      default = "${pkgs.coreutils}/bin/env";
+      example = literalExample ''
+        "''${pkgs.busybox}/bin/env"
```

Comment:
See https://github.com/NixOS/nixpkgs/blob/322a69137754aecf8ce9860b55d7c2bedcc7d873/nixos/modules/config/shells-environment.nix#L123-L125 for where I got it from.
It's a literalExample, so the string value I pass has to be a Nix expression, and that expression happens to contain a `$`, [which you escape using `''` inside a multiline string](https://nixos.org/nix/manual/#ssec-values).